katello/katello-change-hostname

```diff
@@ -256,13 +255,14 @@ unless @foreman_proxy_content
   STDOUT.puts "\nUpdating default #{@proxy}"
   default_capsule_id = hammer_cmd("capsule list | grep #{old_hostname} | awk '{ print $1 }'").chomp
   # Incorrect error message is piped to /dev/null, can be removed when http://projects.theforeman.org/issues/18186 is fixed
-  hammer_cmd("capsule update --id #{default_capsule_id} --url https://#{@new_hostname}:9090 --new-name #{@new_hostname} 2> /dev/null")
+  # For the same reason, we can't use the run_cmd method here
+  hammer_cmd("capsule update --id #{default_capsule_id} --url https://#{@new_hostname}:9090 --new-name #{@new_hostname} 2> /dev/null", [0, 65])
```

Comment:
ugg.... this is hacky. This command always errors due to the issue mentioned in the comments above, so we pipe the ouput to `/dev/null/`. I added an exception for the exit code it returns, but this is assuming the exit code is always the same. 

@evgeni @akofink Is there a better way to handle this? And if not, can I rely on the exit code always being the same on this error? The tl;dr of [the bug](http://projects.theforeman.org/issues/18186) is that it will update the capsule successfully but hammer returns an error.

Reply:
So we could do
```
hammer_cmd("capsule update --id #{default_capsule_id} --url https://#{@new_hostname}:9090 --new-name #{@new_hostname} 2> /dev/null || true")
```
here, or keep the 65 exit code as ok.

I think actually keeping 65 is better, as that limits the possibly-accepted errors (and should be stable: https://github.com/theforeman/hammer-cli/blob/master/lib/hammer_cli/exit_codes.rb#L5).

Just for me to understand: it thinks the new URL is invalid, as at the point you are executing this, the service was not yet really moved, right?